Remove non required db requests

Loki3000 opened the pull request:
Remove SQL requests with empty UID for guests.

Loki3000 committed:
remove non required db requests

Loki3000 committed:
remove non required db requests

MorrisJobke commented on lib/private/Group/Database.php:
```diff
@@ -202,8 +202,11 @@ public function removeFromGroup( $uid, $gid ) {
+		
 		$this->fixDI();
-
+		
```
There should be no space at the end of a line.

MorrisJobke commented on lib/private/Group/Database.php:
```diff
@@ -202,8 +202,11 @@ public function removeFromGroup( $uid, $gid ) {
 	 */
 	public function getUserGroups( $uid ) {
+		//guests has empty $uid
+		if (empty($uid)) return [];
```
Please always use curly brackets.

Loki3000 committed:
fix brackets

Loki3000 committed:
fix spaces

nickvergessen commented on lib/private/Group/Database.php:
```diff
@@ -202,6 +202,11 @@ public function removeFromGroup( $uid, $gid ) {
 	 */
 	public function getUserGroups( $uid ) {
+		//guests has empty $uid
+		if (empty($uid)) {
```
Also falsely matches user with id `'0'`

nickvergessen commented on lib/private/User/Database.php:
```diff
@@ -239,6 +238,12 @@ public function checkPassword($uid, $password) {
 	private function loadUser($uid) {
 		if (!isset($this->cache[$uid])) {
+			//guests $uid could be NULL or ''
+			if (empty($uid)) {
```
Also falsely matches user with id `'0'`

Loki3000 replied:
I have asked here  [https://github.com/nextcloud/server/pull/2998#discussion_r95245581](url) about $uid=='0', but has no answer. So I was suppose it's not allowed.

Loki3000 committed:
allowed '0' uid

Loki3000 committed:
allowed '0' uid

MorrisJobke commented on lib/private/Group/Database.php:
```diff
@@ -202,6 +202,11 @@ public function removeFromGroup( $uid, $gid ) {
 	 */
 	public function getUserGroups( $uid ) {
+		//guests has empty or null $uid
+		if ($uid===null || $uid==='') {
```
please use spaces around the `===`.

MorrisJobke commented on lib/private/User/Database.php:
```diff
@@ -239,6 +238,12 @@ public function checkPassword($uid, $password) {
 	private function loadUser($uid) {
 		if (!isset($this->cache[$uid])) {
+			//guests $uid could be NULL or ''
+			if ($uid===null || $uid==='') {
```
please use spaces around the `===`.

Loki3000 committed:
spaces added

Loki3000 committed:
spaces added

Loki3000 committed:
remove non required sql requests

```
SELECT `appid`, `configkey`, `configvalue` FROM `preferences` WHERE `userid` = '';
```

rullzer commented:
@Loki3000 could you squash your commits?

Loki3000 commented:
@rullzer sorry, I don't know how to do it.
According to [documentation](https://help.github.com/articles/about-pull-request-merges/) it could be made only by repository owner.

rullzer commented:
@Loki3000 see for example http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

rullzer reviewed (approved):
Ah lets just get this in actually! THNX A LOT @Loki3000
